core/lib/dal/src/transactions_dal.rs

```diff
@@ -1752,6 +1752,36 @@ impl TransactionsDal<'_, '_> {
         Ok(rows.len())
     }
 
+    /// Resets `in_mempool` to `FALSE` for the given transaction hashes.
+    pub async fn reset_mempool_status(&mut self, transaction_hashes: &[H256]) -> DalResult<()> {
+        // Convert H256 hashes into `&[u8]`
+        let hashes: Vec<_> = transaction_hashes.iter().map(H256::as_bytes).collect();
+
+        // Execute the UPDATE query
+        let result = sqlx::query!(
+            r#"
+            UPDATE transactions
+            SET
+                in_mempool = FALSE
+            WHERE
+                hash = ANY($1)
+            "#,
+            &hashes as &[&[u8]]
+        )
+        .instrument("return_to_mempool#update_to_false")
```

Comment:
Nit: The name is probably an artifact; `reset_mempool_status` would work better.